eZ/Bundle/EzPublishCoreBundle/Composer/ScriptHandler.php

```diff
@@ -25,12 +25,13 @@ public static function dumpAssets(CommandEvent $event)
         $options = self::getOptions($event);
         $appDir = $options['symfony-app-dir'];
         $webDir = $options['symfony-web-dir'];
-        $env = isset($options['ezpublish-asset-dump-env']) ? $options['ezpublish-asset-dump-env'] : '';
+        // @deprecated: Use of ezpublish-asset-dump-env is deprecated, use standard SYMFONY_ENV instead.
+        $env = !empty($options['ezpublish-asset-dump-env']) ? $options['ezpublish-asset-dump-env'] : (getenv('SYMFONY_ENV') ?: '');
 
         if (!$env) {
             $env = $event->getIO()->ask(
-                "<question>Which environment would you like to dump production assets for?</question> (Default: 'prod', type 'none' to skip) ",
-                'prod'
+                "<question>'SYMFONY_ENV' environment variable not defined, which environment would you like to dump production assets for?</question> (Default: 'dev', type 'none' to skip) ",
+                'dev'
```

Comment:
This, as well as switch to dev as default below is separate commit here as means to better solve https://jira.ez.no/browse/EZP-23159 so that composer.json can again generate assets on composer update.

This is also needed since composer 1.0.0-beta2, `composer install` executes `post-update-cmd` instead of `post-install-cmd`, when there is no composer.lock file: https://github.com/composer/composer/releases/tag/1.0.0-beta2